### DISCUSSION TITLE
iOS 14 widgets

### DISCUSSION THREAD

**Discussion by tylerclark** [ionic-team/capacitor#3175] [Q&A] [answered]
I’m building an app that would be perfect to have some of those iOS widgets on the home screen pages. Any idea how to go about incorporating that into my app?

**Comment by ikeith** [comment, marked as answer]
The home screen widgets are built with [WidgetKit](https://developer.apple.com/documentation/widgetkit/) and, unfortunately, that appears to be out of scope for Capacitor. The widgets themselves are comprised of SwiftUI views (and a timeline of archived data states) which the system manages. There is no web view available. From your Capacitor app, you would need a plugin to which you can pass data for rendering. But all the real work would be in transforming that data into a timeline of native data structures and building the UI to display it in a native app extension. That's all highly specific to your app.

**Comment by balajiks-dev** [comment]
Hi,
The Widgets in iOS14 by SwiftUI is very good easy to build as we have it on the previous state.
We should provide different sizes of views for Widgets like Small, Medium, and Large. 
Check my repo here: https://github.com/balajiks-dev/SwiftUIWidget
